python/triton/language/core.py

```diff
@@ -556,14 +556,14 @@ def __repr__(self):
 
 class pointer_type(dtype):
 
-    def __init__(self, element_ty: dtype, address_space: int = 1):
+    def __init__(self, element_ty: dtype, address_space: int = 1, is_const: bool = False):
         element_ty = _unwrap_if_constexpr(element_ty)
         if not isinstance(element_ty, dtype):
             raise TypeError(f'element_ty has type `{type(element_ty).__name__}`; expected `dtype`.')
         self.element_ty = element_ty
         self.address_space = address_space
-
-        self.name = f'pointer<{element_ty}>'
+        self.is_const = is_const
```

Comment:
is it legal to use `is_const` as it is already the name of the function?

Reply:
yea sorry I converted this back to draft. I need to fix this. Thanks for catching this though!